ONSdigital/census-contact-centre-service
# Cr 1002

# Motivation and Context

CC must not record an invalid address event for a CE
Instead of publishing an invalid address event, the adviser will refer the training guide.

Given the caseType = CE
Then the CCSvc must not publish the address invalid event

The task includes cucumber test

The service should send back a BAD_REQUEST response with the following message in it :

"All CE addresses will be validated by a Field Officer. It is not necessary to submit this Invalidation request"

# What has changed

Changes have been made, mainly to the CaseServiceImpl, so that the getCasesByUprn endpoint will:
1. Get the details of the case that it is tasked to invalidate.
2. Check the value of the caseType that comes back from the case service.
3. If the caseType is "CE" then throw an exception and return a BAD_REQUEST response with the error message specified above.

Relevant JUnit tests have also been amended and a new one added.

# How to test?

Run the mock case service and the contact centre service (branch CR-1002 obviously). Then make sure that the following request in Postman returns a BAD_REQUEST error with this message:
"All CE addresses will be validated by a Field Officer. It is not necessary to submit this Invalidation request"

http://localhost:8171/cases/77346443-64ae-422e-9b93-d5250f48a27a/invalidate

should return this:

{
    "error": {
        "code": "BAD_REQUEST",
        "timestamp": "20200707120533441",
        "message": "All CE addresses will be validated by a Field Officer. It is not necessary to submit this Invalidation request."
    }
}

whereas a case id for an HH case, such as in this request:

http://localhost:8171/cases/03f58cb5-9af4-4d40-9d60-c124c5bddf09/invalidate

should return this:

{
    "id": "03f58cb5-9af4-4d40-9d60-c124c5bddf09",
    "dateTime": "2020-07-07T11:05:38.285Z"
}
